pkg/cmd/admin/image/verify-signature.go

```diff
@@ -0,0 +1,366 @@
+package image
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/containers/image/docker/policyconfiguration"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/signature"
+	sigtypes "github.com/containers/image/types"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	verifyImageSignatureLongDesc = templates.LongDesc(`
+	Verifies the imported image signature using the local public key.
+
+	This command verifies if the signature attached to an image is trusted by
+	using the provided public GPG key.
+	Trusted image means that the image signature was signed by a valid GPG key and the image identity
+	provided by the signature content matches with the image.
+	By default, this command will not record a signature condition back to the Image object but only
+	print the verification status to the console.
+
+	To record a new condition, you have to pass the "--confirm" flag.
```

Comment:
> Dump question, but can I verify just single image with this key?

The command works on single *image*. The problem is that the command can verify at most one *signature* at the time. The image can have multiple signatures though.

> Is it possible to specify multiple public keys when validating multiple images?

Currently not but may be nice. I wouldn't make it a requirement for this PR though.

> how can I clear status from other images

I think a script or `oc observe` are your best options.

> Seems odd reading this.

Agree.